File: C#Part2 - Homework/01. Arrays - Homework/05.MaximalIncreasingSequence/MaximalIncreasingSequence.cs
namespace MaximalIncreasingSequence
{
    using System;
    using System.Collections.Generic;

    class MaximalIncreasingSequence
    {
        // Write a program that finds the maximal increasing sequence in an array.

        static void Main()
        {
            Console.WriteLine("Enter a length for the first array:");
            int n = int.Parse(Console.ReadLine());
            int[] array = new int[n];
            
            for (int i = 0; i < array.Length; i++)
            {
                Console.Write("element[{0}]=", i);
                array[i] = int.Parse(Console.ReadLine());
            }
            List<int> currentSequance = new List<int>() { array[0] };
            List<int> bestSequence = new List<int>();
            int currElement = array[0];
            for (int i = 1; i < array.Length; i++)
            {
                if (array[i] > currElement)
                {
                    currentSequance.Add(array[i]);
                    currElement = array[i];
                }
                else
                {
                    currElement = array[i];
                    currentSequance = new List<int>() { array[i] };
                }
                if (currentSequance.Count > bestSequence.Count)
                {
                    bestSequence = currentSequance;
                }
            }
            for (int i = 0; i < bestSequence.Count; i++)
            {
                if (i < bestSequence.Count - 1)
                {
                    Console.Write(bestSequence[i] + ", ");
                }
                else
                {
                    Console.Write(bestSequence[i]);
                }
            }
            Console.WriteLine();
        }
    }

}

File: JavaScript/Homework/Arrays - Homework/Problem 4 Maximal increasing sequence/MaximalIncreasingSequence.js
// Write a script that finds the maximal increasing sequence in an array.

var i,
    len,
    arr = [3, 2, 3, 4, 2, 2, 4],
    bestSeq = 1,
    currSeq = 1,
    bestNum = '',
    currNum = '';

for (i = 0, len = arr.length; i < len - 1; i+=1) {
    if (arr[i] < arr[i + 1]) {
        currSeq++;
        currNum = currNum + arr[i] + ' ';
    } else {
        if (currSeq >= bestSeq) {
            bestSeq = currSeq;
            currNum = currNum + arr[i] + ' ';
            bestNum = currNum;
        }
        currSeq = 1;
        currNum = '';
    }
}
if (currSeq >= bestSeq) {
    currNum += arr[arr.length - 1];
    bestNum = currNum;
}

console.log("Maximal increasing sequance is: " + bestNum);